index.js

```diff
@@ -0,0 +1,39 @@
+import React, { cloneElement, Children } from 'react';
+import { makeDecorator } from '@storybook/addons';
+
+const createFieldsObject = fields => {
+  if (!fields) {
+    return {};
+  }
+  return Object.keys(fields).reduce((acc, key) => {
+    const value = fields[key];
+    const isStringValue = typeof value === 'string';
+    acc[key] = isStringValue ? { value } : value;
+    return acc;
+  }, {});
+};
+
+const StoryContainer = ({ storyProps, children }) => {
+  const childComponents = Children.map(children, child => {
+    return cloneElement(child, {
+      ...storyProps,
+      fields: createFieldsObject(storyProps.fields)
+    });
+  });
+  return <>
+    {childComponents}
+  </>;
+};
+
+export const withSitecoreProps = makeDecorator({
+  name: 'withSitecoreProps',
+  allowDeprecatedUsage: false,
+  wrapper: (getStory, context) => {
+    const story = getStory(context);
```

Comment:
[suggestion] If a fields property is not defined, can you bail and not run any of this code? That would avoid a your code `createFieldsObject` code failing and also unnecessary child mapping on components that don't take in Sitecore fields.